File: src/main/java/handler/InjuryHandler.java
package handler;

import api.ApiCaller;
import api.DataCollector;
import collect.misc.Injuries;

import java.util.Map;

public class InjuryHandler {
    private DataCollector dataCollector = new DataCollector(new ApiCaller());
    Injuries injuries = new Injuries(dataCollector);

    public Map<String, String> handleRequest(Map<String, String> input) {
        String sport = input.get("sport");
        return sport.equals("nfl") ? injuries.getNFLInjuryData() : injuries.getStandardInjuryData(sport);
    }
}

File: src/main/kotlin/handler/InjuryHandler.kt
package handler

import collect.misc.Injuries

class InjuryHandler {
    fun handleRequest(input: Map<String, String>): Map<String, String> {
        val sport = input.getValue("sport")
        return if (sport == "nfl") getNFLInjuryData() else getStandardInjuryData(sport)
    }

    fun getNFLInjuryData(): Map<String, String> {
        return Injuries().getNFLInjuryData()
    }

    fun getStandardInjuryData(sport: String): Map<String, String> {
        return Injuries().getStandardInjuryData(sport)
    }
}